mirascope/rag/types.py

```diff
@@ -1,8 +1,36 @@
 """Base types and abstract interfaces for typing Mirascope RAG."""
-from typing import Any, Callable, Optional, TypeVar
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Optional, TypeVar
 
 from pydantic import BaseModel, ConfigDict
 
+ResponseT = TypeVar("ResponseT", bound=Any)
+
+
+class BaseEmbeddingResponse(BaseModel, Generic[ResponseT], ABC):
+    """A base abstract interface for LLM embedding responses.
+
+    Attributes:
+        response: The original response from whichever model response this wraps.
+    """
+
+    response: ResponseT
+    start_time: float  # The start time of the completion in ms
```

Comment:
comment here seems copied. Do we actually set start/end time?

Reply:
yes for openai embeddings

Reply:
ok, maybe update to say "the start time of embedding the text in ms"?